analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -18,12 +18,89 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Security.Cryptography;
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
 
 public abstract class HashesShouldHaveUnpredictableSaltBase : SymbolicRuleCheck
 {
     protected const string DiagnosticId = "S2053";
     protected const string MessageFormat = "{0}";
-    protected const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
-    protected const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
+
+    private const int SafeSaltSize = 16;
+    private const string MakeSaltUnpredictableMessage = "Make this salt unpredictable.";
+    private const string MakeThisSaltLongerMessage = "Make this salt at least 16 bytes.";
+
+    private static readonly ImmutableArray<MemberDescriptor> CryptographicallyStrongRandomNumberGenerators = ImmutableArray.Create(
```

Comment:
Wha if you delete this array and you just have a method like so:
```csharp
  private static bool IsCryptographicallyStrongRandomNumberGenerator(IInvocationOperationWrapper invocation) =>
      (invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetBytes)) || invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetNonZeroBytes)))
      && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_RandomNumberGenerator);
```

This way we get rid of the local ` IsInvocationToRandomNumberGenerator` too.

Reply:
I kinda like to keep them there as a list, because the name gives a better understanding of why we use these methods.

Reply:
I personally find it more confusing - it's 3 lines versus 6 or 8.
Additionally, here someone while reading has to go up see the collection and go down again - while if you have only one method in the spot that it needs to be read in the right time it's easier to understand.

Reply:
@Tim-Pohlmann WDYT?

Reply:
Zsolt's solution is more extendable. But YAGNI, so I'd prefer Mary's suggestion (especially because they are from the same type).